--- origsrc/nss-3.12.6/mozilla/security/nss/cmd/vfychain/vfychain.c	2009-04-01 15:41:29.000000000 -0500
+++ src/nss-3.12.6/mozilla/security/nss/cmd/vfychain/vfychain.c	2010-06-07 10:09:18.008786800 -0500
@@ -264,13 +264,13 @@ getCert(const char *name, PRBool isAscii
 #define REV_METHOD_INDEX_MAX  4
 
 typedef struct RevMethodsStruct {
-    uint testType;
+    PRUint32 testType;
     char *testTypeStr;
-    uint testFlags;
+    PRUint32 testFlags;
     char *testFlagsStr;
-    uint methodType;
+    PRUint32 methodType;
     char *methodTypeStr;
-    uint methodFlags;
+    PRUint32 methodFlags;
     char *methodFlagsStr;
 } RevMethods;
 
@@ -280,7 +280,7 @@ SECStatus
 parseRevMethodsAndFlags()
 {
     int i;
-    uint testType = 0;
+    PRUint32 testType = 0;
 
     for(i = 0;i < REV_METHOD_INDEX_MAX;i++) {
         /* testType */
@@ -301,7 +301,7 @@ parseRevMethodsAndFlags()
         /* testFlags */
         if (revMethodsData[i].testFlagsStr) {
             char *flagStr = revMethodsData[i].testFlagsStr;
-            uint testFlags = 0;
+            PRUint32 testFlags = 0;
 
             if (PORT_Strstr(flagStr, REVCONFIG_TEST_TESTLOCALINFOFIRST_STR)) {
                 testFlags |= CERT_REV_MI_TEST_ALL_LOCAL_INFORMATION_FIRST;
@@ -314,7 +314,7 @@ parseRevMethodsAndFlags()
         /* method type */
         if (revMethodsData[i].methodTypeStr) {
             char *methodStr = revMethodsData[i].methodTypeStr;
-            uint methodType = 0;
+            PRUint32 methodType = 0;
             
             if (!PORT_Strcmp(methodStr, REVCONFIG_METHOD_CRL_STR)) {
                 methodType = REVCONFIG_METHOD_CRL;
@@ -333,7 +333,7 @@ parseRevMethodsAndFlags()
         /* method flags */
         if (revMethodsData[i].methodFlagsStr) {
             char *flagStr = revMethodsData[i].methodFlagsStr;
-            uint methodFlags = 0;
+            PRUint32 methodFlags = 0;
 
             if (!PORT_Strstr(flagStr, REVCONFIG_METHOD_DONOTUSEMETHOD_STR)) {
                 methodFlags |= CERT_REV_M_TEST_USING_THIS_METHOD;
@@ -363,7 +363,7 @@ SECStatus
 configureRevocationParams(CERTRevocationFlags *flags)
 {
    int i;
-   uint testType = REVCONFIG_TEST_UNDEFINED;
+   PRUint32 testType = REVCONFIG_TEST_UNDEFINED;
    static CERTRevocationTests *revTests = NULL;
    PRUint64 *revFlags;
 
