--- origsrc/nss-3.12.6/mozilla/dbm/include/mcom_db.h	2009-06-04 18:18:50.000000000 -0500
+++ src/nss-3.12.6/mozilla/dbm/include/mcom_db.h	2010-06-07 09:43:06.812919600 -0500
@@ -59,7 +59,7 @@
 #include <sys/byteorder.h>
 #endif
 
-#if defined(__linux) || defined(__BEOS__)
+#if defined(__linux) || defined(__BEOS__) || defined(__CYGWIN__)
 #include <endian.h>
 #ifndef BYTE_ORDER
 #define BYTE_ORDER    __BYTE_ORDER
--- origsrc/nss-3.12.6/mozilla/security/coreconf/arch.mk	2009-06-04 21:14:49.000000000 -0500
+++ src/nss-3.12.6/mozilla/security/coreconf/arch.mk	2010-06-07 09:43:06.825920300 -0500
@@ -249,7 +249,7 @@ endif
 #
 ifeq (CYGWIN_NT,$(findstring CYGWIN_NT,$(OS_ARCH)))
     OS_RELEASE := $(patsubst CYGWIN_NT-%,%,$(OS_ARCH))
-    OS_ARCH = WINNT
+    OS_ARCH = CYGWIN
     ifndef CPU_ARCH
     ifeq (WOW64,$(findstring WOW64,$(OS_RELEASE)))
         OS_RELEASE := $(patsubst %-WOW64,%,$(OS_RELEASE))
--- origsrc/nss-3.12.6/mozilla/security/coreconf/config.mk	2009-08-25 17:35:11.000000000 -0500
+++ src/nss-3.12.6/mozilla/security/coreconf/config.mk	2010-06-07 09:43:06.832920700 -0500
@@ -63,7 +63,7 @@ endif
 #######################################################################
 
 TARGET_OSES = FreeBSD BSD_OS NetBSD OpenUNIX OS2 QNX Darwin BeOS OpenBSD \
-              AIX RISCOS WINNT WIN95 Linux Android
+              AIX RISCOS WINNT WIN95 Linux Android CYGWIN
 
 ifeq (,$(filter-out $(TARGET_OSES),$(OS_TARGET)))
 include $(CORE_DEPTH)/coreconf/$(OS_TARGET).mk
--- origsrc/nss-3.12.6/mozilla/security/coreconf/CYGWIN.mk	1969-12-31 18:00:00.000000000 -0600
+++ src/nss-3.12.6/mozilla/security/coreconf/CYGWIN.mk	2010-06-07 09:43:06.821920100 -0500
@@ -0,0 +1,134 @@
+#
+# ***** BEGIN LICENSE BLOCK *****
+# Version: MPL 1.1/GPL 2.0/LGPL 2.1
+#
+# The contents of this file are subject to the Mozilla Public License Version
+# 1.1 (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+# http://www.mozilla.org/MPL/
+#
+# Software distributed under the License is distributed on an "AS IS" basis,
+# WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License
+# for the specific language governing rights and limitations under the
+# License.
+#
+# The Original Code is the Netscape security libraries.
+#
+# The Initial Developer of the Original Code is
+# Netscape Communications Corporation.
+# Portions created by the Initial Developer are Copyright (C) 1994-2000
+# the Initial Developer. All Rights Reserved.
+#
+# Contributor(s):
+#
+# Alternatively, the contents of this file may be used under the terms of
+# either the GNU General Public License Version 2 or later (the "GPL"), or
+# the GNU Lesser General Public License Version 2.1 or later (the "LGPL"),
+# in which case the provisions of the GPL or the LGPL are applicable instead
+# of those above. If you wish to allow use of your version of this file only
+# under the terms of either the GPL or the LGPL, and not to allow others to
+# use your version of this file under the terms of the MPL, indicate your
+# decision by deleting the provisions above and replace them with the notice
+# and other provisions required by the GPL or the LGPL. If you do not delete
+# the provisions above, a recipient may use your version of this file under
+# the terms of any one of the MPL, the GPL or the LGPL.
+#
+# ***** END LICENSE BLOCK *****
+
+XP_DEFINE  += -DXP_UNIX
+AR          = ar cr $@
+LDOPTS     += -L$(SOURCE_LIB_DIR)
+LIB_PREFIX	= lib
+LIB_SUFFIX	= a
+DLL_PREFIX	= cyg
+DLL_SUFFIX	= dll
+IMPORT_LIB_SUFFIX	= .dll.a
+DYNAMIC_LIB_SUFFIX_FOR_LINKING = $(IMPORT_LIB_SUFFIX)
+PROG_SUFFIX	= .exe
+
+ifdef BUILD_OPT
+	OPTIMIZER  += -O
+	DEFINES    += -UDEBUG -DNDEBUG
+else
+	OPTIMIZER  += -g
+	DEFINES    += -DDEBUG -UNDEBUG -DDEBUG_$(shell whoami)
+endif
+
+ifdef BUILD_TREE
+NSINSTALL_DIR  = $(BUILD_TREE)/nss
+NSINSTALL      = $(BUILD_TREE)/nss/nsinstall
+else
+NSINSTALL_DIR  = $(CORE_DEPTH)/coreconf/nsinstall
+NSINSTALL      = $(NSINSTALL_DIR)/$(OBJDIR_NAME)/nsinstall
+endif
+
+MKDEPEND_DIR    = $(CORE_DEPTH)/coreconf/mkdepend
+MKDEPEND        = $(MKDEPEND_DIR)/$(OBJDIR_NAME)/mkdepend
+MKDEPENDENCIES  = $(OBJDIR_NAME)/depend.mk
+
+####################################################################
+#
+# One can define the makefile variable NSDISTMODE to control
+# how files are published to the 'dist' directory.  If not
+# defined, the default is "install using relative symbolic
+# links".  The two possible values are "copy", which copies files
+# but preserves source mtime, and "absolute_symlink", which
+# installs using absolute symbolic links.
+#   - THIS IS NOT PART OF THE NEW BINARY RELEASE PLAN for 9/30/97
+#   - WE'RE KEEPING IT ONLY FOR BACKWARDS COMPATIBILITY
+####################################################################
+
+INSTALL = $(NSINSTALL) -t
+
+define MAKE_OBJDIR
+if test ! -d $(@D); then rm -rf $(@D); $(NSINSTALL) -D $(@D); fi
+endef
+
+#
+# The default implementation strategy for Cygwin is pthreads
+#
+USE_PTHREADS = 1
+IMPL_STRATEGY = _PTH
+
+CC			= gcc
+CCC			= g++
+RANLIB			= ranlib
+
+DEFAULT_COMPILER = gcc
+
+OS_REL_CFLAGS	= -DCYGWIN -Di386 -D_XOPEN_SOURCE
+CPU_ARCH	= x86
+
+#ifdef BUILD_OPT
+#	OPTIMIZER	= -O2
+#endif
+
+ifdef USE_PTHREADS
+	DEFINES		+= -D_REENTRANT
+endif
+
+ifeq ($(USE_PTHREADS),1)
+OS_PTHREAD = -lpthread 
+endif
+
+OS_CFLAGS		= $(DSO_CFLAGS) $(OS_REL_CFLAGS) $(ARCHFLAG) -ansi -Wall -pipe -D_BSD_SOURCE -DHAVE_STRERROR -I/usr/include/nspr
+OS_LIBS			= $(OS_PTHREAD)
+
+MKSHLIB		= $(CC) -shared -Wl,--enable-auto-image-base $(_GEN_IMPORT_LIB) $(LDFLAGS)
+
+ARCH			= cygwin
+
+DSO_CFLAGS		= -DPIC -DDLL_EXPORT
+DSO_LDOPTS		= -shared $(ARCHFLAG)
+DSO_LDFLAGS		=
+LDFLAGS			+= $(ARCHFLAG) -L/usr/lib/nspr
+
+ifdef LIBRARY_NAME
+	SHARED_LIBRARY = $(OBJDIR)/$(DLL_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION)$(JDK_DEBUG_SUFFIX).$(DLL_SUFFIX)
+	IMPORT_LIBRARY = $(OBJDIR)/$(LIB_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION)$(JDK_DEBUG_SUFFIX)$(IMPORT_LIB_SUFFIX)
+	_GEN_IMPORT_LIB	= -Wl,--out-implib,$(IMPORT_LIBRARY)
+endif
+
+ifndef TARGETS
+	TARGETS = $(LIBRARY) $(SHARED_LIBRARY) $(IMPORT_LIBRARY) $(PROGRAM)
+endif
--- origsrc/nss-3.12.6/mozilla/security/nss/cmd/shlibsign/sign.sh	2009-12-07 19:33:37.000000000 -0600
+++ src/nss-3.12.6/mozilla/security/nss/cmd/shlibsign/sign.sh	2010-06-07 09:43:06.837921000 -0500
@@ -1,5 +1,13 @@
 #!/bin/sh
 case "${3}" in
+CYGWIN*)
+    PATH="${1}/lib:${1}/bin:${PATH}"
+    export PATH
+    ADDON_PATH=${1}/lib:${4}:$ADDON_PATH
+    export ADDON_PATH
+    echo ${2}/shlibsign -v -i ${5}
+    ${2}/shlibsign -v -i ${5}
+    ;;
 WIN*)
     if echo "${PATH}" | grep -c \; >/dev/null; then
         PATH=${1}/lib\;${1}/bin\;${4}\;${PATH}
--- origsrc/nss-3.12.6/mozilla/security/nss/lib/ckfw/builtins/config.mk	2009-06-10 19:55:34.000000000 -0500
+++ src/nss-3.12.6/mozilla/security/nss/lib/ckfw/builtins/config.mk	2010-06-07 09:43:06.841921200 -0500
@@ -43,7 +43,6 @@ CONFIG_CVS_ID = "@(#) $RCSfile: config.m
 
 TARGETS        = $(SHARED_LIBRARY)
 LIBRARY        =
-IMPORT_LIBRARY =
 PROGRAM        =
 
 ifeq (,$(filter-out WIN%,$(OS_TARGET)))
--- origsrc/nss-3.12.6/mozilla/security/nss/lib/freebl/config.mk	2009-04-11 16:18:42.000000000 -0500
+++ src/nss-3.12.6/mozilla/security/nss/lib/freebl/config.mk	2010-06-07 09:43:06.846921500 -0500
@@ -69,7 +69,6 @@ else
 
 TARGETS      = $(SHARED_LIBRARY)
 LIBRARY      =
-IMPORT_LIBRARY =
 PROGRAM      =
 
 ifeq ($(OS_TARGET), SunOS)
--- origsrc/nss-3.12.6/mozilla/security/nss/lib/freebl/unix_rand.c	2009-06-09 19:42:17.000000000 -0500
+++ src/nss-3.12.6/mozilla/security/nss/lib/freebl/unix_rand.c	2010-06-07 09:43:06.854922000 -0500
@@ -188,7 +188,7 @@ static SECStatus RNG_kstat(PRUint32* fed
 
 #if defined(SCO) || defined(UNIXWARE) || defined(BSDI) || defined(FREEBSD) \
     || defined(NETBSD) || defined(DARWIN) || defined(OPENBSD) \
-    || defined(NTO) || defined(__riscos__)
+    || defined(NTO) || defined(__riscos__) || defined(CYGWIN)
 #include <sys/times.h>
 
 #define getdtablesize() sysconf(_SC_OPEN_MAX)
--- origsrc/nss-3.12.6/mozilla/security/nss/lib/nss/nssinit.c	2010-01-21 20:10:54.000000000 -0600
+++ src/nss-3.12.6/mozilla/security/nss/lib/nss/nssinit.c	2010-06-07 22:57:44.189909500 -0500
@@ -276,6 +276,8 @@ static const char *dllname =
 	"libnssckbi.sl";
 #elif defined(DARWIN)
 	"libnssckbi.dylib";
+#elif defined(CYGWIN)
+	"cygnssckbi.dll";
 #elif defined(XP_UNIX) || defined(XP_BEOS)
 	"libnssckbi.so";
 #else
