--- origsrc/nss-3.16.4/nss/cmd/shlibsign/sign.sh	2014-08-18 16:19:03.585537400 -0500
+++ src/nss-3.16.4/nss/cmd/shlibsign/sign.sh	2014-08-18 15:21:58.000000000 -0500
@@ -11,6 +11,14 @@
 # 5: full path to library that is to be signed
 
 case "${3}" in
+CYGWIN*)
+    PATH="${1}/lib:${1}/bin:${PATH}"
+    export PATH
+    ADDON_PATH=${1}/lib:${4}:$ADDON_PATH
+    export ADDON_PATH
+    echo ${2}/shlibsign -v -i ${5}
+    ${2}/shlibsign -v -i ${5}
+    ;;
 WIN*)
     if echo "${PATH}" | grep -c \; >/dev/null; then
         PATH=${1}/lib\;${1}/bin\;${4}\;${PATH}
--- origsrc/nss-3.16.4/nss/coreconf/CYGWIN.mk	1969-12-31 18:00:00.000000000 -0600
+++ src/nss-3.16.4/nss/coreconf/CYGWIN.mk	2014-08-18 16:18:49.876796600 -0500
@@ -0,0 +1,145 @@
+#
+# ***** BEGIN LICENSE BLOCK *****
+# Version: MPL 1.1/GPL 2.0/LGPL 2.1
+#
+# The contents of this file are subject to the Mozilla Public License Version
+# 1.1 (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+# http://www.mozilla.org/MPL/
+#
+# Software distributed under the License is distributed on an "AS IS" basis,
+# WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License
+# for the specific language governing rights and limitations under the
+# License.
+#
+# The Original Code is the Netscape security libraries.
+#
+# The Initial Developer of the Original Code is
+# Netscape Communications Corporation.
+# Portions created by the Initial Developer are Copyright (C) 1994-2000
+# the Initial Developer. All Rights Reserved.
+#
+# Contributor(s):
+#
+# Alternatively, the contents of this file may be used under the terms of
+# either the GNU General Public License Version 2 or later (the "GPL"), or
+# the GNU Lesser General Public License Version 2.1 or later (the "LGPL"),
+# in which case the provisions of the GPL or the LGPL are applicable instead
+# of those above. If you wish to allow use of your version of this file only
+# under the terms of either the GPL or the LGPL, and not to allow others to
+# use your version of this file under the terms of the MPL, indicate your
+# decision by deleting the provisions above and replace them with the notice
+# and other provisions required by the GPL or the LGPL. If you do not delete
+# the provisions above, a recipient may use your version of this file under
+# the terms of any one of the MPL, the GPL or the LGPL.
+#
+# ***** END LICENSE BLOCK *****
+
+XP_DEFINE  += -DXP_UNIX
+AR          = ar cr $@
+LDOPTS     += -L$(SOURCE_LIB_DIR)
+LIB_PREFIX	= lib
+LIB_SUFFIX	= a
+DLL_PREFIX	= cyg
+DLL_SUFFIX	= dll
+IMPORT_LIB_SUFFIX	= .dll.a
+DYNAMIC_LIB_SUFFIX_FOR_LINKING = $(IMPORT_LIB_SUFFIX)
+PROG_SUFFIX	= .exe
+
+ifdef BUILD_OPT
+	OPTIMIZER  += -O
+	DEFINES    += -UDEBUG -DNDEBUG
+else
+	OPTIMIZER  += -g
+	DEFINES    += -DDEBUG -UNDEBUG -DDEBUG_$(shell whoami)
+endif
+
+ifdef BUILD_TREE
+NSINSTALL_DIR  = $(BUILD_TREE)/nss
+NSINSTALL      = $(BUILD_TREE)/nss/nsinstall
+else
+NSINSTALL_DIR  = $(CORE_DEPTH)/coreconf/nsinstall
+NSINSTALL      = $(NSINSTALL_DIR)/$(OBJDIR_NAME)/nsinstall
+endif
+
+MKDEPEND_DIR    = $(CORE_DEPTH)/coreconf/mkdepend
+MKDEPEND        = $(MKDEPEND_DIR)/$(OBJDIR_NAME)/mkdepend
+MKDEPENDENCIES  = $(OBJDIR_NAME)/depend.mk
+
+####################################################################
+#
+# One can define the makefile variable NSDISTMODE to control
+# how files are published to the 'dist' directory.  If not
+# defined, the default is "install using relative symbolic
+# links".  The two possible values are "copy", which copies files
+# but preserves source mtime, and "absolute_symlink", which
+# installs using absolute symbolic links.
+#   - THIS IS NOT PART OF THE NEW BINARY RELEASE PLAN for 9/30/97
+#   - WE'RE KEEPING IT ONLY FOR BACKWARDS COMPATIBILITY
+####################################################################
+
+INSTALL = $(NSINSTALL) -t
+
+define MAKE_OBJDIR
+if test ! -d $(@D); then rm -rf $(@D); $(NSINSTALL) -D $(@D); fi
+endef
+
+#
+# The default implementation strategy for Cygwin is pthreads
+#
+USE_PTHREADS = 1
+IMPL_STRATEGY = _PTH
+
+CC			= gcc
+CCC			= g++
+RANLIB			= ranlib
+
+DEFAULT_COMPILER = gcc
+
+OS_REL_CFLAGS	= -DCYGWIN
+
+ifeq ($(OS_TEST),x86_64)
+	CPU_ARCH        = x86_64
+	USE_64		= 1
+else
+	OS_REL_CFLAGS   += -Di386
+	CPU_ARCH        = x86
+endif
+
+#ifdef BUILD_OPT
+#	OPTIMIZER	= -O2
+#endif
+
+ifdef USE_PTHREADS
+	DEFINES		+= -D_REENTRANT
+endif
+
+ifeq ($(USE_PTHREADS),1)
+OS_PTHREAD = -lpthread 
+endif
+
+OS_CFLAGS		= $(DSO_CFLAGS) $(OS_REL_CFLAGS) $(ARCHFLAG) -Wall -pipe -DHAVE_STRERROR -I/usr/include/nspr
+OS_LIBS			= $(OS_PTHREAD)
+
+MKSHLIB		= $(CC) -shared -Wl,--enable-auto-image-base $(_GEN_IMPORT_LIB) $(LDFLAGS)
+
+ARCH			= cygwin
+
+DSO_CFLAGS		= -DPIC -DDLL_EXPORT
+DSO_LDOPTS		= -shared $(ARCHFLAG)
+DSO_LDFLAGS		=
+LDFLAGS			+= $(ARCHFLAG)
+
+ifdef LIBRARY_NAME
+	SHARED_LIBRARY = $(OBJDIR)/$(DLL_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION)$(JDK_DEBUG_SUFFIX).$(DLL_SUFFIX)
+	IMPORT_LIBRARY = $(OBJDIR)/$(LIB_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION)$(JDK_DEBUG_SUFFIX)$(IMPORT_LIB_SUFFIX)
+	_GEN_IMPORT_LIB	= -Wl,--out-implib,$(IMPORT_LIBRARY)
+endif
+
+ifndef TARGETS
+	TARGETS = $(LIBRARY) $(SHARED_LIBRARY) $(IMPORT_LIBRARY) $(PROGRAM)
+endif
+
+USE_SYSTEM_ZLIB = 1
+ZLIB_LIBS       = -lz
+
--- origsrc/nss-3.16.4/nss/coreconf/arch.mk	2014-08-18 16:19:03.594038500 -0500
+++ src/nss-3.16.4/nss/coreconf/arch.mk	2014-08-18 15:21:58.000000000 -0500
@@ -190,7 +190,7 @@ endif
 #
 ifeq (CYGWIN_NT,$(findstring CYGWIN_NT,$(OS_ARCH)))
     OS_RELEASE := $(patsubst CYGWIN_NT-%,%,$(OS_ARCH))
-    OS_ARCH = WINNT
+    OS_ARCH = CYGWIN
     ifndef CPU_ARCH
     ifeq (WOW64,$(findstring WOW64,$(OS_RELEASE)))
         OS_RELEASE := $(patsubst %-WOW64,%,$(OS_RELEASE))
@@ -199,7 +199,7 @@ ifeq (CYGWIN_NT,$(findstring CYGWIN_NT,$
 	#
 	# Cygwin's uname -m returns "i686" on a Pentium Pro machine.
 	#
-	ifneq (,$(findstring 86,$(CPU_ARCH)))
+	ifneq (,$(findstring i686,$(CPU_ARCH)))
 	    CPU_ARCH = x386
 	endif
     endif
--- origsrc/nss-3.16.4/nss/coreconf/config.mk	2014-08-18 16:19:03.600539300 -0500
+++ src/nss-3.16.4/nss/coreconf/config.mk	2014-08-18 15:21:58.000000000 -0500
@@ -31,7 +31,7 @@ endif
 #######################################################################
 
 TARGET_OSES = FreeBSD BSD_OS NetBSD OpenUNIX OS2 QNX Darwin BeOS OpenBSD \
-              AIX RISCOS WINNT WIN95 Linux Android
+              AIX RISCOS WINNT WIN95 Linux Android CYGWIN
 
 ifeq (,$(filter-out $(TARGET_OSES),$(OS_TARGET)))
 include $(CORE_DEPTH)/coreconf/$(OS_TARGET).mk
--- origsrc/nss-3.27.1/nss/external_tests/common/gtest.mk	2016-10-03 09:55:58.000000000 -0500
+++ src/nss-3.27.1/nss/external_tests/common/gtest.mk	2016-10-07 13:03:56.714941400 -0500
@@ -7,7 +7,7 @@
 include $(CORE_DEPTH)/cmd/platlibs.mk
 
 MKPROG = $(CCC)
-MKSHLIB	= $(CCC) $(DSO_LDOPTS) $(DARWIN_SDK_SHLIBFLAGS)
+MKSHLIB	= $(CCC) $(DSO_LDOPTS) $(DARWIN_SDK_SHLIBFLAGS) $(_GEN_IMPORT_LIB)
 
 # gtests pick up errors with signed/unsigned comparisons on some platforms
 # even though we disabled -Wsign-compare.
@@ -32,5 +32,5 @@ ifeq (WINNT,$(OS_ARCH))
     # Needed because we include files from a subdirectory
     MAKE_OBJDIR = $(INSTALL) -D $(dir $@)
 else
-    CXXFLAGS += -std=c++0x
+    CXXFLAGS += -std=c++0x -D_POSIX_C_SOURCE=200809L
 endif
--- origsrc/nss-3.16.4/nss/lib/Makefile	2014-08-05 12:24:47.000000000 -0500
+++ src/nss-3.16.4/nss/lib/Makefile	2014-08-18 16:11:48.398275700 -0500
@@ -37,10 +37,8 @@ endif
 endif
 
 ifndef MOZILLA_CLIENT
-ifeq ($(OS_ARCH),Linux)
 SYSINIT_SRCDIR = sysinit  # Add the sysinit directory to DIRS.
 endif
-endif
 
 ifndef NSS_DISABLE_DBM
 DBM_SRCDIR = dbm  # Add the dbm directory to DIRS.
--- origsrc/nss-3.16.4/nss/lib/ckfw/builtins/config.mk	2014-08-18 16:19:03.605039900 -0500
+++ src/nss-3.16.4/nss/lib/ckfw/builtins/config.mk	2014-08-18 15:21:58.000000000 -0500
@@ -10,7 +10,6 @@
 
 TARGETS        = $(SHARED_LIBRARY)
 LIBRARY        =
-IMPORT_LIBRARY =
 PROGRAM        =
 
 ifeq (,$(filter-out WIN%,$(OS_TARGET)))
--- origsrc/nss-3.16.4/nss/lib/freebl/config.mk	2014-08-18 16:19:03.609540500 -0500
+++ src/nss-3.16.4/nss/lib/freebl/config.mk	2014-08-18 15:21:58.000000000 -0500
@@ -38,7 +38,6 @@ else
 
 TARGETS      = $(SHARED_LIBRARY)
 LIBRARY      =
-IMPORT_LIBRARY =
 PROGRAM      =
 
 ifeq ($(OS_TARGET), SunOS)
--- origsrc/nss-3.27.1/nss/lib/freebl/unix_rand.c
+++ src/nss-3.27.1/nss/lib/freebl/unix_rand.c
@@ -160,7 +160,7 @@ static SECStatus RNG_kstat(PRUint32* fed
 
 #endif
 
-#if defined(SCO) || defined(UNIXWARE) || defined(BSDI) || defined(FREEBSD) || defined(NETBSD) || defined(DARWIN) || defined(OPENBSD) || defined(NTO) || defined(__riscos__)
+#if defined(SCO) || defined(UNIXWARE) || defined(BSDI) || defined(FREEBSD) || defined(NETBSD) || defined(DARWIN) || defined(OPENBSD) || defined(NTO) || defined(__riscos__) || defined(__CYGWIN__)
 #include <sys/times.h>
 
 #define getdtablesize() sysconf(_SC_OPEN_MAX)
